src/architecture/utilities/orbitalMotion.c

```diff
@@ -285,24 +236,18 @@ double H2N(double H, double e)
  */
 double M2E(double M, double e)
 {
+    assert(0 <= e && e < 1);
     double small = 1e-13;
     double dE = 10 * small;
     double E1 = M;
-    int    max = 200;
-    int    count = 0;
-
-    if((e >= 0) && (e < 1)) {
-        while(fabs(dE) > small) {
-            dE = (E1 - e * sin(E1) - M) / (1 - e * cos(E1));
-            E1 -= dE;
-            if(++count > max) {
-                BSK_PRINT(MSG_ERROR, "iteration error in M2E(%f,%f)", M, e);
-                dE = 0.;
-            }
-        }
-    } else {
-        E1 = NAN;
-        BSK_PRINT(MSG_ERROR, "M2E() received e = %g. The value of e should be 0 <= e < 1.", e);
+    int    maxIterations = 200;
+    int    iterationCount = 0;
+
+    while(fabs(dE) > small) {
+        assert(iterationCount <= maxIterations);
```

Comment:
Looking more closely at how `maxIterations` is used, I think it might not be appropriate to use `assert` here.

It seems that the original logic was just to bail out of the loop if `maxIterations` was reached, and return the current approximation of `E1`. The original code used `dE = 0.;` as an implicit `break`. since the looping condition fails if `dE` isn't a large enough step size.

I would recommend changing the code to read as:

```
    for (int iterationCount = 0; iterationCount <= maxIterations; ++iterationCount) {
        if (fabs(dE) <= small) break;
        dE = (E1 - e * sin(E1) - M) / (1 - e * cos(E1));
        E1 -= dE;
    }
```

Reply:
Good call. 👍